lib/codegen/src/isa/mod.rs

```diff
@@ -129,6 +129,15 @@ pub enum LookupError {
     Unsupported,
 }
 
+impl fmt::Display for LookupError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            LookupError::SupportDisabled => write!(f, "Support for this target is disabled"),
+            LookupError::Unsupported => write!(f, "Support for this target is not implemented yet"),
+        }
+    }
+}
```

Comment:
We've started to use the `#[fail(...)]` macros in a few places for this kind of code, such as [here](https://github.com/CraneStation/cranelift/blob/master/lib/codegen/src/result.rs#L14), although I don't know if being slightly more concise is worth the benefit of being somewhat more magical. Do you have a preference?

Reply:
Yeah, it's more concise, I like it better, thanks!